libpod/container_internal_linux.go

```diff
@@ -1442,6 +1448,24 @@ func (c *Container) makeBindMounts() error {
 		}
 	}
 
+	// Bind notify proxy
+	if c.config.SdNotifyMode == define.SdNotifyModeContainer {
+		notifyDir := filepath.Join(c.bundlePath(), "notify")
+		logrus.Debugf("checking notify %q dir", notifyDir)
+		if err := os.MkdirAll(notifyDir, 0755); err != nil {
+			if !os.IsExist(err) {
+				return errors.Wrapf(err, "unable to create notify %q dir", notifyDir)
+			}
+		}
+		if err := label.Relabel(notifyDir, c.MountLabel(), true); err != nil {
+			return errors.Wrapf(err, "relabel failed %q", notifyDir)
+		}
+		logrus.Debugf("add bindmount notify %q dir", notifyDir)
+		if _, ok := c.state.BindMounts["/run/notify"]; !ok {
+			c.state.BindMounts["/run/notify"] = notifyDir
+		}
+	}
+
```

Comment:
similarly here, we need the OCI runtime to take care of it.  If we pass down the notify socket directly to the container, there is a race condition between the notification being sent and systemd reading it.

Reply:
@giuseppe We're proxying. PODMAN is now in charge of notification.  NOT the OCI runtime. That's the point.

Reply:
PODMAN creates /run/notify/
PODMAN tells OCI runtime to bind-mount /run/notify to the container's setup folder
CONMON forwards /run/notify/notify.sock ( in container, by way of bind mount ) to --sdnotify-socket on host

OCI runtime is NOT taking care of it.